ci/tlk.ps1

```diff
@@ -329,65 +383,97 @@ class TlkRecipe
+                $DestinationSymbolsName = $(Split-Path $DestinationExecutable -Leaf).Replace(".$($this.Target.ExecutableExtension)", ".$($this.Target.SymbolsExtension)")
+                $DestinationDirectory  = Split-Path $DestinationExecutable -Parent
+                Copy-Item $SrcSymbolsPath -Destination $(Join-Path $DestinationDirectory $DestinationSymbolsName)
+            } elseif (!$this.Target.IsWindows()) {
+                $StripExecutable = 'strip'
+                if (Test-Path Env:STRIP_EXECUTABLE) {
+                    $StripExecutable = $Env:STRIP_EXECUTABLE
                 }
+    
+                & $StripExecutable $SrcBinaryPath | Out-Host
             }
-            Default {
-                $null
+    
+            if ($DestinationExecutable) {
+                Copy-Item -Path $SrcBinaryPath -Destination $DestinationExecutable
             }
-        }
 
-        if ($this.Target.IsWindows() -And $DestinationExecutable) {
-            $SrcSymbolsName = $CargoPackage.Replace('-','_')
-            $SrcSymbolsName = $SrcSymbolsName, $this.Target.SymbolsExtension -ne '' -Join '.'
-            $SrcSymbolsPath = "$($this.SourcePath)/target/${CargoTarget}/${CargoProfile}/${SrcSymbolsName}"
-            $DestinationSymbolsName = $(Split-Path $DestinationExecutable -Leaf).Replace(".$($this.Target.ExecutableExtension)", ".$($this.Target.SymbolsExtension)")
-            $DestinationDirectory  = Split-Path $DestinationExecutable -Parent
-            Copy-Item $SrcSymbolsPath -Destination $(Join-Path $DestinationDirectory $DestinationSymbolsName)
-        } elseif (!$this.Target.IsWindows()) {
-            $StripExecutable = 'strip'
-            if (Test-Path Env:STRIP_EXECUTABLE) {
-                $StripExecutable = $Env:STRIP_EXECUTABLE
+            if ($CargoPackage.Name -Eq 'devolutions-pedm-contextmenu') {
+                if ($Null -Eq (Get-Command "MakeAppx.exe" -ErrorAction SilentlyContinue) -Or $Null -Eq (Get-Command "SignTool.exe" -ErrorAction SilentlyContinue)) {
+                    throw 'MakeAppx was not found in the PATH'
+                }
+
+                $CargoPackagePath = Get-Location
+
+                Push-Location
+                Set-Location $CargoOutputPath
+    
+                $MakeAppxOutput = & 'MakeAppx.exe' 'pack' '/f' "${CargoPackagePath}/mapping.txt" '/p' "./devolutions-pedm-contextmenu.msix" '/nv' '/o'
+                if (!$?) {
+                    throw "MakeAppx package creation failed: ${MakeAppxOutput}"
+                }
+
+                if (-Not ($(Test-Path $Env:DAGENT_PEDM_PFX) -And $Env:DAGENT_PEDM_PFX_PASSWORD)) {
+                    throw 'DAGENT_PEDM_PFX / DAGENT_PEDM_PFX_PASSWORD was not found in the environment'
+                }
+
+                $SignToolOutput = & 'SignTool.exe' 'sign' '/fd' 'SHA256' '/a' '/f' "${Env:DAGENT_PEDM_PFX}" '/p' "${Env:DAGENT_PEDM_PFX_PASSWORD}" (Get-Item "./devolutions-pedm-contextmenu.msix").FullName
```

Comment:
Yes, signing needs to be done in the CI workflow. You can invoke devops to get the proper certificates/secrets made available. 

It's ok to allow signing in tlk for dev/test purposes with a local certificate, but not for releases.

Signing is normally only done in the package workflow since privileged access to the signing information requires the workflow run to be authorized. If it's important that CI artifacts are also signed, we'd need to consider options here (perhaps sign in CI with a dev / low privilege certificate, and then resign in packaging)